addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -0,0 +1,925 @@
+        JsonAutomation[] result = parseJson(json, JsonAutomation[].class);
+        return result;
+    }
+
+    public JsonFeed[] getEnabledFlashBriefings() throws IOException, URISyntaxException {
+        String json = makeRequestAndReturnString(alexaServer + "/api/content-skills/enabled-feeds");
+        JsonEnabledFeeds result = parseJson(json, JsonEnabledFeeds.class);
+        JsonFeed[] enabledFeeds = result.enabledFeeds;
+        if (enabledFeeds != null) {
+            return enabledFeeds;
+        }
+        return new JsonFeed[0];
+    }
+
+    public void setEnabledFlashBriefings(JsonFeed[] enabledFlashBriefing) throws IOException, URISyntaxException {
+        JsonEnabledFeeds enabled = new JsonEnabledFeeds();
+        enabled.enabledFeeds = enabledFlashBriefing;
+        GsonBuilder gsonBuilder = new GsonBuilder();
+        gsonBuilder.serializeNulls();
+        Gson gson = gsonBuilder.create();
+        String json = gson.toJson(enabled);
+        makeRequest("POST", alexaServer + "/api/content-skills/enabled-feeds", json, true, true, null);
+    }
+
+    public JsonNotificationSound[] getNotificationSounds(Device device) throws IOException, URISyntaxException {
+        String json = makeRequestAndReturnString(
+                alexaServer + "/api/notification/sounds?deviceSerialNumber=" + device.serialNumber + "&deviceType="
+                        + device.deviceType + "&softwareVersion=" + device.softwareVersion);
+        JsonNotificationSounds result = parseJson(json, JsonNotificationSounds.class);
+        JsonNotificationSound[] notificationSounds = result.notificationSounds;
+        if (notificationSounds != null) {
+            return notificationSounds;
+        }
+        return new JsonNotificationSound[0];
+    }
+
+    public JsonNotificationResponse notification(Device device, String type, @Nullable String label,
+            @Nullable JsonNotificationSound sound) throws IOException, URISyntaxException {
+        Date date = new Date(new Date().getTime());
+        long createdDate = date.getTime();
+        Date alarm = new Date(createdDate + 5000); // add 5 seconds, because amazon does not except calls for times in
+                                                   // the past (compared with the server time)
+        long alarmTime = alarm.getTime();
+
+        JsonNotificationRequest request = new JsonNotificationRequest();
+        request.type = type;
+        request.deviceSerialNumber = device.serialNumber;
+        request.deviceType = device.deviceType;
+        request.createdDate = createdDate;
+        request.alarmTime = alarmTime;
+        request.reminderLabel = label;
+        request.sound = sound;
+        request.originalDate = new SimpleDateFormat("yyyy-MM-dd").format(alarm);
+        request.originalTime = new SimpleDateFormat("HH:mm:ss.SSSS").format(alarm);
+        request.type = type;
+        request.id = "create" + type;
+
+        GsonBuilder gsonBuilder = new GsonBuilder();
```

Comment:
This is potentially expensive - you should instantiate it only once as a field in that class.